01_task_block.py

```diff
@@ -0,0 +1,177 @@
+
+"""
+=============================================================
+Extract segments of the data recorded during task performance
+=============================================================
+Segments that were recorded during the self-paced breaks (in between
+experimental blocks) will be dropped.
+Authors: José C. García Alanis <alanis.jcg@gmail.com>
+License: BSD (3-clause)
+"""
+import matplotlib.pyplot as plt
+
+from mne.io import read_raw_fif
+from mne import events_from_annotations, concatenate_raws, open_report
+
+# All parameters are defined in config.py
+from config import fname, n_jobs, parser, LoggingFormat
+
+# Handle command line arguments
+args = parser.parse_args()
+subject = args.subject
+
+print(LoggingFormat.PURPLE +
+      LoggingFormat.BOLD +
+      'Extracting task blocks for subject %s' % subject +
+      LoggingFormat.END)
+
+###############################################################################
+# 1) Import the output from previous processing step
+input_file = fname.output(subject=subject,
+                          processing_step='raw_files',
+                          file_type='raw.fif')
+raw = read_raw_fif(input_file, preload=True)
+
+# drop status channel
+raw.drop_channels('Status')
+
+###############################################################################
+# 2) Find periods of time in the data with no presented stimuli (i.e., the
+# self-paced breaks)
+
+# relevant events
+ids = {'245': 1,  # end of block
+       '71': 2,   # flanker onset
+       '11': 3,   # target_L_C
+       '12': 4,   # target_R_C
+       '21': 5,   # target_L_I
+       '22': 6,   # target_R_I
+       '101': 7,  # correct_incongruent
+       '102': 8,  # correct_congruent
+       '201': 9,  # incorrect_incongruent
+       '202': 10  # incorrect_congruent
+       }
+
+# extract events
+events = events_from_annotations(raw, event_id=ids)
+
+# cue events
+cue_evs = events[0]
+cue_evs = cue_evs[(cue_evs[:, 2] >= 7) & (cue_evs[:, 2] <= 12)]
+
+# latencies and difference between two consecutive cues
+latencies = cue_evs[:, 0] / raw.info['sfreq']
+diffs = [(y - x) for x, y in zip(latencies, latencies[1:])]
+
+# get first event after a long break (i.e., when the time difference between
+# stimuli is greater than 10 seconds). This should only be the case in between
+# task blocks.
+breaks = [diff for diff in range(len(diffs)) if diffs[diff] > 10]
+print('\n Identified breaks at positions', breaks)
```

Comment:
I don’t believe its necessary to identify breaks in this experiment. We didn't record eeg during self-paced breaks in the task, remember? The only thing we should think about is to separate the practice trials from the main task. Practice trials should be the first 40 trials or so. So let’s just get rid of those.